cadc-vos-server-alt/src/main/java/org/opencadc/vospace/server/async/RecursiveDeleteNodeRunner.java

```diff
@@ -0,0 +1,358 @@
+                    // nothing to do here
+                }
+            } else {
+                try {
+                    nodePersistence.delete(serverNode);
+                    log.debug("Recursively deleted data node " + nodePath);
+                } catch (Exception ex) {
+                    log.debug("Cannot recursively delete node " + nodePath, ex);
+                    incErrorCount();
+                }
+            }
+
+            // set the appropriate end phase and error summary
+            ErrorSummary error = null;
+            ExecutionPhase endPhase = ExecutionPhase.COMPLETED;
+            if (deleteCount == 0) {
+                endPhase = ExecutionPhase.ERROR;
+                ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.EXECUTING, endPhase, error, new Date());
+            } else {
+                List<Result> results = new ArrayList<>();
+                results.add(new Result("delcount", URI.create("final:" + deleteCount)));
+                if (errorCount > 0) {
+                    endPhase = ExecutionPhase.ABORTED;
+                    results.add(new Result("errorcount", URI.create("final:" + errorCount)));
+                    ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.EXECUTING, endPhase, results, new Date());
+                } else {
+                    ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.EXECUTING, endPhase, results, new Date());
+                }
+            }
+            if (!endPhase.equals(ep)) {
+                log.warn("Could not change the job phase from " + ExecutionPhase.EXECUTING + " to " + endPhase);
+            }
+        } catch (FileNotFoundException e) {
+            sendError("NotFound");
+        } catch (AccessControlException e) {
+            sendError("PermissionDenied");
+        } catch (Throwable t) {
+            log.error("Unexpected exception", t);
+            // if the cause of this throwable is an InterruptedException, and if
+            // the job has been aborted, then this can be considered a normal
+            // abort procedure
+            if (t instanceof InterruptedException || ThrowableUtil.isACause(t, InterruptedException.class)) {
+                try {
+                    ExecutionPhase ep = jobUpdater.getPhase(job.getID());
+                    if (ExecutionPhase.ABORTED.equals(ep)) {
+                        return;
+                    }
+                } catch (Exception e) {
+                    log.error("Could not check job phase: ", e);
+                }
+            }
+
+            sendError("Unexpected Exception: " + t.getMessage());
+        }
+    }
+
+    private boolean deleteContainer(ContainerNode node, Subject caller) throws Exception {
+
+        boolean errors = false;
+        List<ContainerNode> childContainers = new ArrayList<>();
+        log.debug("Deleting container " + Utils.getPath(node));
+        ResourceIterator<Node> iterator = nodePersistence.iterator(node, null, null);
```

Comment:
this needs to use `try-with-resources` so that `close()` is automatically and always called

Reply:
Done. Good catch